// gatsby-plugin-prefetch-google-fonts/gatsby-ssr.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var React = require("react");

var globby = require("globby").sync;

var _require = require("fs"),
    readFileSync = _require.readFileSync;

exports.onRenderBody = function (_ref) {
  var setHeadComponents = _ref.setHeadComponents;
  var files = globby("./public/google-fonts/**/*.woff2");
  var css = readFileSync("./.cache/google-fonts/google-fonts.css");
  setHeadComponents((0, _toConsumableArray2.default)(files.map(function (file, key) {
    return React.createElement("link", {
      key: "googleFont".concat(key),
      rel: "preload",
      as: "font",
      type: "font/woff2",
      crossOrigin: "anonymous",
      href: file.replace("./public", "")
    });
  })).concat([React.createElement("style", {
    key: "googleFontsCSS",
    type: "text/css",
    dangerouslySetInnerHTML: {
      __html: css
    }
  })]));
};
